server/service/src/invoice/inventory_adjustment/adjust_existing_stock/generate.rs

```diff
@@ -79,45 +94,62 @@ pub fn generate(
         cost_price_per_pack,
         sell_price_per_pack,
         note,
+        on_hold,
         ..
     } = stock_line.stock_line_row.clone();
 
-    let invoice_line = InvoiceLineRow {
-        id: uuid(),
-        invoice_id: invoice.id.clone(),
-        item_link_id: stock_line.item_row.id,
-        item_name: stock_line.item_row.name,
-        item_code: stock_line.item_row.code,
-        stock_line_id: Some(stock_line_id),
-        location_id,
-        batch,
-        expiry_date,
-        pack_size,
-        cost_price_per_pack,
-        sell_price_per_pack,
-        total_before_tax: 0.0,
-        total_after_tax: 0.0,
-        tax_percentage: None,
-        r#type: match adjustment_type {
-            AdjustmentType::Addition => InvoiceLineType::StockIn,
-            AdjustmentType::Reduction => InvoiceLineType::StockOut,
-        },
-        number_of_packs: adjustment,
-        note,
-        inventory_adjustment_reason_id,
-        return_reason_id: None,
-        foreign_currency_price_before_tax: None,
-    };
+    let invoice_id = invoice.id.clone();
+    let invoice_line_id = uuid();
 
-    let mut updated_stock_line = stock_line.stock_line_row;
-
-    let delta = match adjustment_type {
-        AdjustmentType::Addition => adjustment,
-        AdjustmentType::Reduction => -adjustment,
+    let insert_stock_in_or_out_line = match adjustment_type {
+        AdjustmentType::Addition => InsertStockInOrOutLine::StockIn(InsertStockInLine {
+            r#type: StockInType::InventoryAddition,
+            id: invoice_line_id.clone(),
+            invoice_id,
+            stock_line_id: Some(stock_line_id),
+            number_of_packs: adjustment,
+            // From existing stock line
+            item_id: stock_line.item_row.id,
+            location: location_id.map(|id| NullableUpdate { value: Some(id) }),
```

Comment:
location: None (nullalbe update is made to be able to ignore the update if it's nullable but not specified)

Reply:
I think most of these can be None (to use existing stock line's value ?)